src/CleanRegex/Internal/Prepared/Cluster/FigureClusters.php

```diff
@@ -12,19 +12,24 @@ class FigureClusters implements CountedClusters
     /** @var string[] */
     private $figures;
 
+    /**
+     * @param string[] $figures
+     */
     public function __construct(array $figures)
     {
         foreach ($figures as $figure) {
             if (\is_string($figure)) {
                 continue;
             }
+            // @phpstan-ignore-next-line Don't trust library user's input
             throw InvalidArgument::typeGiven("Invalid inject figure type. Expected string", new ValueType($figure));
         }
         $this->figures = \array_slice($figures, 0);
     }
 
     public function current(): Cluster
     {
+        // @phpstan-ignore-next-line FIXME: \current can return false
         return new AtomicGroup(new LiteralFigure(\current($this->figures)));
```

Comment:
Is this a false-positive (could this method be called when the $figures is empty)? I haven't analysed the flow yet.

Reply:
I am impressed by the verbosity of the answer BTW, thank you for taking time to explain.